### DISCUSSION TITLE
Custom JDBC Connection URL parameters

### DISCUSSION THREAD

**Discussion by yaroslavambrozyak** [apache/kyuubi#1783] [Q&A] [answered]
Hi, I am aware that we can provide some configs using connection parameters like 
`jdbc:hive2://localhost:10009/default;#spark.sql.shuffle.partitions=2;spark.executor.memory=5g`

In my case, I don't want users to pass specific parameters manually but want to create a few profiles that will contain predefined configs
For instance:
I want to create 2 profiles like **SMALL** and **LARGE**.
They will contain some predefined configs, let's say 
**SMALL** will have predefined 
`spark.executor.memory=1g`  
**LARGE** will have predefined
`spark.executor.memory=5g`

So then the user can pass the profile via URL parameters like 
`jdbc:hive2://localhost:10009/default;#env.profile=SMALL` 
and using this parameter Kyuubi will create instances with different environments set up (in this case it will apply `spark.executor.memory=1g`)

Is this doable? Thanks.

**Comment by wForget** [comment, marked as answer]
May be related to these issues:
https://github.com/apache/incubator-kyuubi/issues/966#issuecomment-901945984
https://github.com/apache/incubator-kyuubi/issues/1536

**Reply by yaroslavambrozyak** [reply]
Thanks, #1536 is exactly what I was looking for

**Comment by yaooqinn** [comment]
thanks @yaroslavambrozyak for sharing your idea. #925 can you share your use case here？it makes sense to me and a real world case may be more convincing to others too.

**Reply by yaroslavambrozyak** [reply]
Sure, what actually do you want me to share?

**Reply by yaooqinn** [reply]
simply follow the example if appropriate
